cmd/ld-relay/ld-relay.go

```diff
@@ -38,15 +39,35 @@ func main() {
 		logging.Error.Printf("Error initializing metrics: %s", err)
 	}
 
-	err = http.ListenAndServe(fmt.Sprintf(":%d", c.Main.Port), r)
+	if c.Main.TLSEnabled {
+		if c.Main.TLSCert == "" {
+			logging.Error.Printf("TLS: tlsCert required")
+			os.Exit(1)
+		}
+		if c.Main.TLSKey == "" {
+			logging.Error.Printf("TLS: tlsKey required")
+			os.Exit(1)
+		}
+
+		go func() {
+			err = http.ListenAndServeTLS(fmt.Sprintf(":%d", c.Main.Port), c.Main.TLSCert, c.Main.TLSKey, r)
+		}()
+	} else {
+		go func() {
+			err = http.ListenAndServe(fmt.Sprintf(":%d", c.Main.Port), r)
+		}()
+	}
+
 	if err != nil {
 		if c.Main.ExitOnError {
 			logging.Error.Fatalf("Error starting http listener on port: %d  %s", c.Main.Port, err)
 		}
 		logging.Error.Printf("Error starting http listener on port: %d  %s", c.Main.Port, err)
 	} else {
-		logging.Info.Printf("Listening on port %d\n", c.Main.Port)
+		logging.Info.Printf("TLS Enabled: %s, Listening on port %d\n", strconv.FormatBool(c.Main.TLSEnabled), c.Main.Port)
```

Comment:
I'm not crazy about this wording: I know what you mean by "`TLS Enabled: false`" but at first glance it makes it look like TLS _is_ enabled. Since this option will probably be rarely used, I'd rather just not mention TLS at all if it's not enabled.

Reply:
I can make it an additional logged line if it is enabled, does that work? I was trying to figure out how to add it as unobtrusively as possible but for it to be logged.

Reply:
I think an extra line is fine.